example/sample.ts

```diff
@@ -34,12 +34,8 @@ const IMAGE_PATH = '/samantha-gades-BlIhVfXbi9s-unsplash.jpg';
 const state = {
   defaultDevices: new Map<MediaDeviceKind, string>(),
   bitrateInterval: undefined as any,
-  blur: BackgroundBlur(BLUR_RADIUS, undefined, (stats) => {
-    // console.log('frame processing stats', stats);
-  }),
-  virtualBackground: VirtualBackground(IMAGE_PATH, undefined, (stats) => {
-    // console.log('frame processing stats', stats);
-  }),
+  backgroundMode: null as NonNullable<BackgroundProcessorOptions['mode']> | null,
```

Comment:
Instead of tracking this within the sample, how about holding the current mode (and exposing a way to retrieve it) on the `BackgroundProcessorWrapper` directly?

Reply:
I thought about doing this, but the reason I didn't was because if I did the bookkeeping in `BackgroundProcessorWrapper`, then somebody could just call `updateTransformerOptions` themselves and get it into a broken state.

The concern you are bringing up is valid though and I think it would make the logic much cleaner - let me think about how to best accomplish this. Maybe I could override `updateTransformerOptions` and do that check at that level to guard against this 🤔

Reply:
I thought about it and added a `mode` getter which inspects the options and computes the mode on the fly. I think that's the least intrusive option!